### PR TITLE
Changing `round` to use the new arg count feature

### PR DESCRIPTION
**Pull request by abenkovskii**
solves #993 
P.S. In my view removing an old `functionsOverloads` should be a separate refactoring pass

### CODE REVIEW

File: src/kOS.Safe/Compilation/KS/Compiler.cs [hunk 1]
```diff
@@ -36,10 +36,8 @@ class Compiler
         private CompilerOptions options;
         private const bool TRACE_PARSE = false; // set to true to Debug Log each ParseNode as it's visited.
 
-        private readonly Dictionary<string, string> functionsOverloads = new Dictionary<string, string>
+        private readonly Dictionary<string, string> functionsOverloads = new Dictionary<string, string> // TODO: remove this
```
Comment (erendrake) [inline src/kOS.Safe/Compilation/KS/Compiler.cs]: if we dont need it, remove it.

**Comment by erendrake** [comment]
I am fine with removing unused features like this with the same PR, but you are welcome to do it in two of course.

**Comment by Dunbaratu** [comment]
In this case the feature didn't BECOME unused until this PR removed the need for it, so it makes sense for its removal to be part of this PR.  Prior to this PR's changes to the round() function, that piece of code was still needed.

**Comment by erendrake** [comment]
@Dunbaratu yeah, that would be my preference too. @abenkovskii if you pull out the feature in this PR or another ill merge them at the same time.

**Comment by abenkovskii** [comment]
OK. I'll try to figure out what parts of the compiler are unused now and can safely be removed and add commits to this PR. Should I test all the `kerboscript_tests/` every time I modify the compiler?

**Comment by erendrake** [comment]
@abenkovskii i just sent you a PR that makes a few changes, including the compiler changes. i didnt have a chance to test it yet.

**Pull request merged**